src/frontend/komponenter/Fagsak/Behandlingsresultat/EndretUtbetalingAvslagBegrunnelse.tsx

```diff
@@ -0,0 +1,153 @@
+        const begrunnelser = lastedeTekster
+            ? lastedeTekster[type]
+                  .filter(begrunnelse =>
+                      begrunnelse.endringsårsaker.includes(
+                          IEndretUtbetalingAndelÅrsak.ALLEREDE_UTBETALT
+                      )
+                  )
+                  .map(begrunnelse => ({
+                      label: begrunnelse.navn,
+                      value: begrunnelse.id,
+                  }))
+            : [];
+        return {
+            label: begrunnelseTyper[type],
+            options: begrunnelser,
+        };
+    });
+
+    if (endretUtbetalingsbegrunnelser.status === RessursStatus.FEILET) {
+        return (
+            <Alert variant="error">
+                Klarte ikke å hente inn begrunnelser for endret utbetaling.
+            </Alert>
+        );
+    }
+
+    const finnBegrunnelseType = (begrunnelse: Begrunnelse): BegrunnelseType | undefined => {
+        return Object.keys(lastedeTekster).find(begrunnelseType => {
+            return lastedeTekster
+                ? lastedeTekster[begrunnelseType as BegrunnelseType].find(
+                      (begrunnelseTilknyttetVilkår: IRestBegrunnelseTilknyttetEndretUtbetaling) =>
+                          begrunnelseTilknyttetVilkår.id === begrunnelse
+                  ) !== undefined
+                : '';
+        }) as BegrunnelseType;
+    };
+
+    function finnBegrunnelseForSelect(begrunnelseVerdi?: Begrunnelse) {
+        if (!lastedeTekster) return;
+
+        for (const key in lastedeTekster) {
+            const valgtBegrunnelse = lastedeTekster[key as BegrunnelseType].find(begrunnelse => {
+                return begrunnelse.id === begrunnelseVerdi;
+            });
+
+            if (valgtBegrunnelse) {
+                return {
+                    label: valgtBegrunnelse.navn,
+                    value: valgtBegrunnelse.id,
+                };
+            }
+        }
+    }
+
+    return (
+        <FamilieReactSelect
+            {...skjema.felter.begrunnelser.hentNavInputProps(skjema.visFeilmeldinger)}
+            value={finnBegrunnelseForSelect(prevalgtBegrunnelse)}
+            label={'Velg standardtekst i brev'}
+            creatable={false}
+            placeholder={'Velg begrunnelse'}
+            erLesevisning={vurderErLesevisning()}
+            isMulti={false}
+            options={grupperteBegrunnelser}
+            onChange={options => {
+                if (options && 'value' in options) {
+                    /* 
+                        Vi ønsker kun en begrunnelse her, men modellen tilsier at det skal være en array.
+                        Hardkoder dermed den ene valgte verdien som en array med en begrunnelse.
+                    */
+                    skjema.felter.begrunnelser.validerOgSettFelt([options.value]);
+                } else {
+                    skjema.felter.begrunnelser.validerOgSettFelt([]);
+                }
+            }}
```

Comment:
Er det mulig at`onChange` trigges uten at det finnes `options` og en valgt verdi? Trenger vi if'en her? Eller kunne `onChange` bare gjort ```skjema.felter.begrunnelser.validerOgSettFelt([options.value])``` uten if/else. 🤔 Mulig det er noe jeg ikke ser her altså 🙈

Reply:
Vi trenger nok en if av et slag her. Hvis vi ikke sjekker på noe og verdi er undefined, vil vi få en array med en index som også er undefined i stedet for en tom array. 😊